cli/lsp/config.rs

```diff
@@ -192,23 +368,34 @@ impl Config {
     }
   }
 
-  pub fn update_specifier(
-    &mut self,
-    specifier: ModuleSpecifier,
-    value: Value,
-  ) -> LSPResult<()> {
-    let settings: SpecifierSettings = serde_json::from_value(value)
-      .map_err(|err| LSPError::invalid_params(err.to_string()))?;
-    self.specifier_settings.insert(specifier, settings);
+  pub async fn update_specifier_settings(
+    &self,
+    specifier: &ModuleSpecifier,
+    uri: &ModuleSpecifier,
+  ) -> Result<(), AnyError> {
+    if self.client_capabilities.workspace_configuration
+      && !self
+        .settings
+        .try_read()
```

Comment:
When would you expect this to fail?

I'd probably just send unconditionally, without checking .specifiers first, and let the receiver handle it.

Reply:
Yeah, when I was pondering this offline I came to the same thought, it would put an extra message in the queue, but it would get cleared down in a much narrower lock situation.